docs/DeveloperGuide.md

```diff
@@ -1,34 +1,172 @@
+
+The `UI` component,
+
+*   Executes user commands using the `Logic` component
+
+#### Logic component
+The `Logic` component 
+
+1. `Logic` uses the `Parser` class to parse the user command.
+2. The parsed command is passed to `Controller` which then returns a specific command class 
+e.g. `AddCommand`, `FindCommand` etc. which is executed by the main class `Duke`.
+All these command classes inherits from the abstract `Command` class.
+3. The command execution can affect the Model (e.g. adding a module in ModuleList)
+4. The result of the command execution is passed back to the Ui.
+5. In addition, the command execution can also instruct the `Ui` to perform certain actions, 
+such as displaying help to the user.
+
+#### Model component
+The `Model` component
+* Stores a `Person` object that represents the user's details. (e.g. `totalCap`, `totalModuleCreditCompleted`)
+* Stores the `ModuleList` and `SemesterList` Data
+* Does not depend on any of the other three components
+
+#### Storage component
+The `Storage` component,
+* can save `personInfo` objects in csv format and read it back
+* can save the available module list in csv format and read it back
+* can save the semester list in csv format and read it back 
+
+
+### Implementation
+
+#### `Add to Semester` feature 
+The `Add to Semester` mechanism is facilitated by `AddtoSemCommand` which extends from `Command`. 
+It allows `ModuleManager` to assign a module to a semester by adding the module to a 
+`SemModulesList`, it implements the following operations:
+
+Given below is an example how the `Add to Semester` behaves at each step.
+
+##### Step 1:
+The user launches the application for the first time. The `SemesterList` will be initialized with the none 
+`SemModulesList` in it.
+
+##### Step 2:
+When users enter a add to semester command, e.g `add id/CS2113 s/4 mc/4`, this command will be parsed in `Parser`
+and then `Parser` returns a `AddToSemCommand`. `AddToSemCommand` then calls `Command#execute(SemesterList semesterList,
+ AvailableModulesList availableModulesList) `
+`(AddToSemCommand#execute(SemesterList semesterList, AvailableModulesList availableModulesList))`
+
+##### Step 3:
+`AddToSemCommand#execute()` then calls self method `AddToSemCommand#addModule()`.`AddToSemCommand#addModule()`
+ then calls `AddToSemCommand#checkModuleExist(semesterList)` to check whether the selected 
+module is already in the selected module list (i.e:`semesterList`, which is a `PriorityQueue<SemModulesList>`). 
+If the module is not in the list, `AddToSemCommand#addModule()` will check whether there is a semester list
+(i.e:`semesterModulesList`, which is a `ArrayList<SelectedModule>`) whose name is the module's semester name. 
+If the semester list exist, the module will be added to the list. 
+If not, `AddToSemCommand#addModule()` will create a new semester list and then add this module to the new list. and the
+the new semester list will be added to `semesterList` as well.
```

Comment:
I found this section quite confusing to read as there are a few variables and classes with similar names (semesterList, SemModulesList, semesterModulesList, SemesterList), and the multiple 'if' condition checks. Perhaps it would help to briefly explain the connection between semesterList and semesterModulesList, at the very beginning, before delving into the logic? I.e. "semesterList represents XXX and stores instances of SemModuleList. semesterModulesList represents XXX and stores XXX"